packages/babel-parser/src/parser/statement.ts

```diff
@@ -222,6 +222,7 @@ export default abstract class StatementParser extends ExpressionParser {
         this.raise(Errors.ModuleExportUndefined, { at, localName });
       }
     }
+    this.eat(tt.eof); // stop before } when parsing a module expression
```

Comment:
Why do we need to eat it?

Reply:
Can we move it as `tt.expect` in `parseTopLevel`, to assert that we actually consumed all the tokens?

Reply:
If we eat `tt.eof` in `parseTopLevel`, the `Program` will be ended early than it should be and as a result its trailing comments won't append. I revises it a bit and now it only consume when `end` is `tt.eof`, which must be invoked by `parseTopLevel`.